nemo/collections/tts/data/audio_trimming.py

```diff
@@ -0,0 +1,292 @@
+               vad_threshold: Softmax probability [0, 1] of VAD output, above which audio frames will be classified
+                 as speech.
+               device: Device "cpu" or "cuda" to use for running the VAD model.
+               frame_length: Length of audio frames to use when doing speech detection. This does not need to match
+                 the frame_length used any other part of the code or model.
+               frame_step: Stride of audio frames to use when doing speech detection. This does not need to match
+                 the frame_step used any other part of the code or model.
+               pad_seconds: Amount of audio to keep before the detected start of speech and after the end of
+                 speech. Set this to at least 0.1 to avoid cutting off any speech audio, with larger values
+                 being safer but increasing the amount of silence left afterwards.
+               volume_norm: Whether to normalize the volume of audio before doing speech detection.
+        """
+        self.device = device
+        self.vad_model = EncDecClassificationModel.from_pretrained(model_name=model_name).eval().to(self.device)
+        self.vad_sample_rate = vad_sample_rate
+        self.vad_threshold = vad_threshold
+
+        self.frame_threshold = frame_threshold
+        self.frame_length = frame_length
+        self.frame_step = frame_step
+
+        self.pad_seconds = pad_seconds
+        self.volume_norm = volume_norm
+
+    def _detect_speech(self, audio: np.array) -> np.array:
+        # Center-pad the audio
+        audio = np.pad(audio, [self.frame_length // 2, self.frame_length // 2])
+
+        # [num_frames, frame_length]
+        audio_frames = librosa.util.frame(
+            audio, frame_length=self.frame_length, hop_length=self.frame_step
+        ).transpose()
+
+        num_frames = audio_frames.shape[0]
+        # [num_frames, frame_length]
+        audio_signal = torch.tensor(audio_frames, dtype=torch.float32, device=self.device)
+        # [1]
+        audio_signal_len = torch.tensor(num_frames * [self.frame_length], dtype=torch.int32, device=self.device)
+
+        # VAD outputs 2 values for each audio frame with logits indicating the likelihood that
+        # each frame is non-speech or speech, respectively.
+        # [num_frames, 2]
+        log_probs = self.vad_model(input_signal=audio_signal, input_signal_length=audio_signal_len)
+        probs = torch.softmax(log_probs, dim=-1)
+        probs = probs.cpu().detach().numpy()
+        # [num_frames]
+        speech_probs = probs[:, 1]
+        speech_frames = speech_probs >= self.vad_threshold
+
+        return speech_frames
+
+    def _scale_sample_indices(self, start_sample_i: int, end_sample_i: int, sample_rate: int) -> Tuple[int, int]:
+        sample_rate_ratio = sample_rate / self.vad_sample_rate
+        start_sample_i = sample_rate_ratio * start_sample_i
+        end_sample_i = sample_rate_ratio * end_sample_i
```

Comment:
pad_sample_indices() called after this is what handles the bounding and integer conversion. I could also bound the min/max and convert the final output to integer within this function too if it makes things clearer.

Reply:
Ah gotcha, I see it. Either is fine, though if it's kept as-is can you put in a note for the future in case anyone wants to call it by itself for whatever reason?

Reply:
+1 with above suggestion.